util/generate_compilation_db_test.py

```diff
@@ -0,0 +1,270 @@
+    args_after_compiler = ['-o', 'out', 'src.c', '-iquote', 'include-path',
+                           'some-other-arg']
+
+    def test_not_gcc(self):
+        args_not_gcc = ['cc'] + self.args_after_compiler
+        action = BazelAqueryAction({'arguments': args_not_gcc})
+        action.hackily_hardcode_googletest_includes()
+        self.assertEqual(action.arguments, args_not_gcc)
+
+    def test_simple(self):
+        args_with_gcc = ['gcc'] + self.args_after_compiler
+        action = BazelAqueryAction({'arguments': args_with_gcc})
+        action.hackily_hardcode_googletest_includes()
+        self.assertEqual(action.arguments,
+                         ['gcc', '-o', 'out', 'src.c',
+                          '-iquote', 'bazel-opentitan/external/googletest/googletest/include',
+                          '-iquote', 'bazel-opentitan/external/googletest/googlemock/include',
+                          '-iquote', 'include-path', 'some-other-arg'])
+
+
+# A pared-down example of Bazel aquery output. Generated with `./bazelisk.sh
+# aquery --output=jsonproto '//sw/device/lib/crypto:otbn_util'`.
+BAZEL_AQUERY_RESULTS_SMALL = r"""
+{
+  "artifacts": [{
+    "id": 1,
+    "pathFragmentId": 1
+  }, {
+    "id": 2,
+    "pathFragmentId": 6
+  }, {
+    "id": 3,
+    "pathFragmentId": 11
+  }, {
+    "id": 4,
+    "pathFragmentId": 12
+  }, {
+    "id": 5,
+    "pathFragmentId": 15
+  }, {
+    "id": 6,
+    "pathFragmentId": 16
+  }],
+  "actions": [{
+    "targetId": 1,
+    "actionKey": "709e80c88487a2411e1ee4dfb9f22a861492d20c4765150c0c794abd70f8147c",
+    "mnemonic": "Middleman",
+    "configurationId": 1,
+    "inputDepSetIds": [1],
+    "outputIds": [3],
+    "primaryOutputId": 3,
+    "executionPlatform": "@local_config_platform//:host"
+  }, {
+    "targetId": 1,
+    "actionKey": "e0abcf3f57dcd54d61576eb49b6a4911ed9fc6af72d3dd61548d6e396e8736c4",
+    "mnemonic": "CppCompile",
+    "configurationId": 1,
+    "arguments": ["/usr/bin/gcc", "-Wall", "-iquote", ".", "-isystem", "external/googletest/googlemock", "-fno-canonical-system-headers", "-c", "sw/device/lib/crypto/otbn_util.c", "-o", "bazel-out/k8-fastbuild/bin/sw/device/lib/crypto/_objs/otbn_util/otbn_util.pic.o"],
+    "inputDepSetIds": [2],
+    "outputIds": [7, 8],
+    "discoversInputs": true,
+    "primaryOutputId": 7,
+    "executionPlatform": "@local_config_platform//:host"
+  }],
+  "targets": [{
+    "id": 1,
+    "label": "//sw/device/lib/crypto:otbn_util",
+    "ruleClassId": 1
+  }],
+  "depSetOfFiles": [{
+    "id": 1,
+    "directArtifactIds": [1, 2]
+  }, {
+    "id": 3,
+    "directArtifactIds": [4, 5, 6]
+  }, {
+    "id": 2,
+    "transitiveDepSetIds": [3],
+    "directArtifactIds": [3]
+  }],
+  "configuration": [{
+    "id": 1,
+    "mnemonic": "k8-fastbuild",
+    "platformName": "k8",
+    "checksum": "b617ac78fd0cd265dcf958077ac7aae01530f6896f0958308502ccf9026482e6"
+  }],
+  "ruleClasses": [{
+    "id": 1,
+    "name": "cc_library"
+  }],
+  "pathFragments": [{
```

Comment:
The `pathFragments` list could definitely be pared down, but I'm not sure it's worth the effort. This list contains unreachable elements because I removed a bunch of values from `depSetOfFiles` that were unnecessary for testing.